src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java

```diff
@@ -0,0 +1,200 @@
+    }
+    Schema schema = ObjectType.fromString(objectName, collector).getObjectSchema();
+    specBuilder.setSchema(schema);
+    return specBuilder.addRelatedPlugin(
+      new PluginSpec(ZendeskBatchSource.NAME, BatchSource.PLUGIN_TYPE, properties)).build();
+  }
+
+  @Override
+  public List<StructuredRecord> sample(ConnectorContext connectorContext, SampleRequest sampleRequest)
+    throws IOException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    List<StructuredRecord> recordList = new ArrayList<>();
+    objectType = ObjectType.fromString(sampleRequest.getPath(), collector);
+    Schema schema = objectType.getObjectSchema();
+    for (String record : getResponseAsMap()) {
+      recordList.add(StructuredRecordStringConverter.fromJsonString(record, schema));
+    }
+    return recordList;
+  }
+
+  List<String> getResponseAsMap() throws IOException {
+    subdomain = config.getSubdomains().stream().findFirst().get();
+    String baseUrl = String.format(ZendeskConstants.BASE_URL, subdomain, objectType.getApiEndpoint());
+    if (objectType.isBatch()) {
+      // start time = current date - 30 days
+      long epochTime = ZonedDateTime.now().toEpochSecond() - (30 * 24 * 60 * 60);
+      baseUrl = String.format(baseUrl.contains("?") ? "%s&%s" : "%s?%s", baseUrl,
+        String.format("start_time=%s", epochTime));
+    }
+    if (objectType == ObjectType.ARTICLE_COMMENTS || objectType == ObjectType.POST_COMMENTS) {
+      return getCommentsObjectMap(ObjectType.USERS_SIMPLE).stream().map(t -> getJsonValuesFromResponse(t,
+        objectType)).flatMap(t -> t.stream()).collect(Collectors.toList());
+    } else if (objectType == ObjectType.REQUESTS_COMMENTS) {
+      return getCommentsObjectMap(ObjectType.REQUESTS).stream().map(t -> getJsonValuesFromResponse(t,
+        objectType)).flatMap(t -> t.stream()).collect(Collectors.toList());
+    }
+    return getJsonValuesFromResponse(getResponseMap(baseUrl), objectType);
+  }
+
+  // Fetches the response for the selected object in Zendesk.
+  private Map<String, Object> getResponseMap(String baseUrl) throws IOException {
+    CloseableHttpClient httpClient = HttpUtil.createHttpClient(config);
+    HttpClientContext httpClientContext = HttpUtil.createHttpContext(config, baseUrl);
+    //replace out %2B with + due to API restriction
+    URI uri = URI.create(ZendeskConstants.RESTRICTED_PATTERN.matcher(baseUrl).replaceAll("+"));
+    try (CloseableHttpResponse response = httpClient.execute(new HttpGet(uri), httpClientContext)) {
+      int statusCode = response.getStatusLine().getStatusCode();
+      if (statusCode / 100 == 2) {
+        String responseAsString = new String(EntityUtils.toByteArray(response.getEntity()),
+          StandardCharsets.UTF_8);
+        return (Map<String, Object>) GSON.fromJson(responseAsString, Map.class);
+      }
+      throw new HttpResponseException(statusCode, "No response.");
```

Comment:
done

Reply:
Log the response body

Reply:
please fix

Reply:
You should log the response string.
Something like this:
```
      LOG.error("Unable to fetch the response: {}", responseAsString);
```
Please make sure to get an internal review for these changes

Reply:
Done, with the internal review.

Reply:
Change
```      LOG.error("Unable to fetch the response.", responseAsString);
```
to:
```
      LOG.error("Unable to fetch the response: {}", responseAsString);
```
Otherwise responseAsString won't be logged

Reply:
done